dipy/reconst/tests/test_qtdmri.py

```diff
@@ -0,0 +1,583 @@
+import numpy as np
+from dipy.data import get_gtab_taiwan_dsi
+from numpy.testing import (assert_,
+                           assert_almost_equal,
+                           assert_array_almost_equal,
+                           assert_equal,
+                           assert_raises,
+                           run_module_suite)
+from dipy.reconst import qtdmri, mapmri
+from dipy.sims.voxel import MultiTensor
+from dipy.data import get_sphere
+from dipy.sims.voxel import add_noise
+import scipy.integrate as integrate
+from dipy.core.gradients import gradient_table_from_qvals_bvecs
+
+
+def generate_gtab4D(number_of_tau_shells=4, delta=0.01):
+    """Generates testing gradient table for 4D qt-dMRI scheme"""
+    gtab = get_gtab_taiwan_dsi()
+    qvals = np.tile(gtab.bvals / 100., number_of_tau_shells)
+    bvecs = np.tile(gtab.bvecs, (number_of_tau_shells, 1))
+    pulse_separation = []
+    for ps in np.linspace(0.02, 0.05, number_of_tau_shells):
+        pulse_separation = np.append(pulse_separation,
+                                     np.tile(ps, gtab.bvals.shape[0]))
+    pulse_duration = np.tile(delta, qvals.shape[0])
+    gtab_4d = gradient_table_from_qvals_bvecs(qvals=qvals, bvecs=bvecs,
+                                              big_delta=pulse_separation,
+                                              small_delta=pulse_duration)
+    return gtab_4d
+
+
+def generate_signal_crossing(gtab, lambda1, lambda2, lambda3, angle2=60):
+    mevals = np.array(([lambda1, lambda2, lambda3],
+                       [lambda1, lambda2, lambda3]))
+    angl = [(0, 0), (angle2, 0)]
+    S, sticks = MultiTensor(gtab, mevals, S0=1.0, angles=angl,
+                            fractions=[50, 50], snr=None)
+    return S
+
+
+def test_input_parameters():
+    gtab_4d = generate_gtab4D()
+    try:
+        qtdmri.QtdmriModel(gtab_4d, radial_order=3)
+        assert_equal(True, False)
+    except ValueError:
+        print('uneven radial_order is caught')
+
+    try:
+        qtdmri.QtdmriModel(gtab_4d, radial_order=-1)
+        assert_equal(True, False)
+    except ValueError:
+        print('negative radial_order is caught')
```

Comment:
same as above multiple time. Can you update all try/except below?

Reply:
Changed.